lib/workers/repository/process/index.ts

```diff
@@ -1,24 +1,67 @@
 import { mergeChildConfig } from '../../../config';
 import type { RenovateConfig } from '../../../config/types';
+import { CONFIG_VALIDATION } from '../../../constants/error-messages';
 import { logger } from '../../../logger';
 import type { PackageFile } from '../../../manager/types';
+import { platform } from '../../../platform';
+import { getCache } from '../../../util/cache/repository';
 import { branchExists } from '../../../util/git';
 import { addSplit } from '../../../util/split';
 import type { BranchConfig } from '../../types';
 import { readDashboardBody } from '../dependency-dashboard';
 import { ExtractResult, extract, lookup, update } from './extract-update';
 import type { WriteUpdateResult } from './write';
 
-function getBaseBranchConfig(
+async function getBaseBranchConfig(
   baseBranch: string,
   config: RenovateConfig
-): RenovateConfig {
+): Promise<RenovateConfig> {
   logger.debug(`baseBranch: ${baseBranch}`);
-  const baseBranchConfig = mergeChildConfig(config, { baseBranch });
-  if (config.baseBranches.length > 1) {
+
+  let baseBranchConfig: RenovateConfig = config;
+
+  if (
+    config.useBaseBranchConfig === 'replace' &&
+    baseBranch !== config.defaultBranch
+  ) {
+    logger.debug(
+      { baseBranch },
+      `Using config from base branch because useBaseBranchConfig option specified`
+    );
+
+    // Retrieve config file name autodetected for this repo
+    const cache = getCache();
+    const configFileName = cache.configFileName;
```

Comment:
Does this mean that the config file name in the baseBranch must match the file name in the default branch? If so then we should document this.